redir.php

```diff
@@ -1,9 +1,36 @@
 <?php
+// URL to list of CSV Shortlinks
+$URLToCSV="https://raw.githubusercontent.com/hyphacoop/shortlinks/master/shortlinks.csv";
 
+// Enable caching of file by defining the number of minutes the file is considered stale
+// 0 = disables caching
+$cacheInterval=15;
+// Path to the file where the caching will be stored
+// File must be writable
+// Hint: chown www-data.www-data shortlinks.csv 
+$cacheFile=dirname(__FILE__) . "/shortlinks.csv";
+
+$MSG="";
 $link = $_REQUEST['link'];
 
 $tolink = "";
-$csv = file_get_contents("https://raw.githubusercontent.com/hyphacoop/shortlinks/master/shortlinks.csv");
+
+if ($cacheInterval) { 
+    if (is_writable($cacheFile)) {
+        if (time()-filemtime($filename) >  $cacheInterval * 60) {
```

Comment:
Download an autoformatter.

```suggestion
        if (time()-filemtime($filename) > $cacheInterval * 60) {
```